@here/harp-mapview/lib/poi/PoiRenderer.ts

```diff
@@ -539,62 +561,64 @@ export class PoiRenderer {
         const imageTextureName = poiInfo.imageTextureName;
 
         const imageTexture = this.mapView.poiManager.getImageTexture(imageTextureName);
-        if (imageTexture === undefined) {
-            // Warn about a missing texture, but only once.
-            if (PoiRenderer.m_missingTextureName.get(imageTextureName) === undefined) {
-                PoiRenderer.m_missingTextureName.set(imageTextureName, true);
-                logger.error(`preparePoi: No imageTexture with name '${imageTextureName}' found`);
-            }
-            poiInfo.isValid = false;
-            return;
-        }
+        let imageItem: ImageItem;
+        let imageCache: MapViewImageCache;
+        if (imageTexture) {
+            const imageDefinition = imageTexture.image;
 
-        const imageDefinition = imageTexture.image;
+            const image = this.mapView.imageCache.findImageByName(imageDefinition);
 
-        // Check user image cache first.
-        let imageItem = this.mapView.userImageCache.findImageByName(imageDefinition);
-        let imageCache: MapViewImageCache | undefined;
-        if (imageItem === undefined) {
-            // Then check default image cache.
-            imageItem = this.mapView.imageCache.findImageByName(imageDefinition);
-            if (imageItem === undefined) {
+            if (!image) {
                 logger.error(`init: No imageItem found with name '${imageDefinition}'`);
                 poiInfo.isValid = false;
                 return;
-            } else {
-                imageCache = this.mapView.imageCache;
             }
+            imageItem = image;
+            imageCache = this.mapView.imageCache;
         } else {
+            // No image texture found. Either this is a user image or it's missing.
+            const image = this.mapView.userImageCache.findImageByName(imageTextureName);
+
+            if (!image) {
+                // Warn about a missing texture, but only once.
+                if (PoiRenderer.m_missingTextureName.get(imageTextureName) === undefined) {
+                    PoiRenderer.m_missingTextureName.set(imageTextureName, true);
+                    logger.error(
+                        `preparePoi: No imageTexture with name '${imageTextureName}' found`
+                    );
+                }
+                poiInfo.isValid = false;
+                return;
```

Comment:
Now the user images don't need to be registered both in MapView's userImageCache and PoiManager. This only makes sense internally for theme atlases.